README.md

```diff
@@ -18,18 +21,245 @@ Add to your SBT dependencies:
 "com.codacy" % "codacy-engine-scala-seed_2.12" % "<VERSION>"
 ```
 
-You shouldn't worry about the library itself, we use it as a core in our tools,
-and everything is well explained in our Docs section.
-
 ## Docs
 
-[Docker Docs](http://docs.codacy.com/v1.5/docs/tool-developer-guide)
+### How to integrate an external analysis tool on Codacy
+
+By creating a docker and writing code to handle the tool invocation and output,
+you can integrate the tool of your choice on Codacy!
+
+> To know more about dockers, and how to write a docker file please refer to [https://docs.docker.com/reference/builder/](https://docs.docker.com/reference/builder/)
+
+In this tutorial, we explain how you can integrate an analysis tool of your choice in Codacy.
+You can check the code of an already implemented tool and, if you wish, fork it to start your own.
+You are free to modify and use it for your own tools.
+
+#### Requirements
+
+* Docker definition with the tool you want to integrate
+* Define the documentation for the patterns provided by the tool
+
+#### Assumptions and Behaviour
+
+* To run the tool we provide the configuration file, `/.codacyrc`, with the language to run and optional parameters your tool might need.
+* The source code to be analysed will be located in `/src`, meaning that when provided in the configuration, the file paths are relative to `/src`.
+
+* **Structure of the .codacyrc file:**
+  * **files:** Files to be analysed (their path is relative to `/src`)
+  * **tools:** Array of tools
+    * **name:** Unique identifier of the tool. This will be provided by the tool in patterns.json file.
+    * **patterns:** Array of patterns that must be checked
+      * **patternId:** Unique identifier of the pattern
+      * **parameters:** Parameters of the pattern
+        * **name:** Unique identifier of the parameter
+        * **value:** Value to be used as parameter value
+
+```json
+{
+  "files" : ["foo/bar/baz.js", "foo2/bar/baz.php"],
+  "tools":[
+    {
+      "name":"jshint",
+      "patterns":[
+        {
+          "patternId":"latedef",
+          "parameters":[
+            {
+              "name":"latedef",
+              "value":"vars"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+Regarding the configuration file, the tool should have different behaviours for the following situations:
```

Comment:
Guys, all these text still seems a little bit complicated. Do you think it would be better to have some kind of pseudo-code?

Reply:
I think this is enough